File: scraper/submission.py
class submission(object):
    def __init__(self, author, clicked, created_utc, distinguished, edited,
            sid, is_original_content, is_self, link_flair_text,
            locked, name, num_comments, over_18, permalink, saved, score,
            selftext, spoiler, stickied, subreddit, subreddit_id, title,
            upvote_ratio, url) -> None:
        self.type = "submission"
        self.author = author
        self.clicked = clicked
        self.created_utc = created_utc
        self.distinguished = distinguished
        self.edited = edited
        self.id = sid
        self.is_original_content = is_original_content
        self.is_self = is_self
        self.link_flair_text = link_flair_text
        self.locked = locked
        self.name = name
        self.num_comments = num_comments
        self.over_18 = over_18
        self.permalink = permalink
        self.submission_link = "https://reddit.com" + permalink
        self.saved = saved
        self.score = score
        self.selftext = selftext
        self.spoiler = spoiler
        self.stickied = stickied
        self.subreddit = "r/" + subreddit
        self.subreddit_id = subreddit_id
        self.title = title
        self.upvote_ratio = upvote_ratio
        self.url = url

File: graphql/src/models/submission.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const submissionSchema = new Schema({
  type: {
    type: String,
    required: true,
  },
  author: {
    type: String,
    required: true,
  },
  clicked: {
    type: Boolean,
    required: true,
  },
  created_utc: {
    type: Number,
    required: true,
  },
  distinguished: {
    type: String,
    required: false,
    default: null,
  },
  edited: {
    type: Boolean,
    required: true,
  },
  reddit_id: {
    type: String,
    required: true,
  },
  is_original_content: {
    type: Boolean,
    required: true,
  },
  is_self: {
    type: Boolean,
    required: true,
  },
  link_flair_text: {
    type: String,
    required: true,
  },
  locked: {
    type: Boolean,
    required: true,
  },
  name: {
    type: String,
    required: true,
  },
  num_comments: {
    type: Number,
    required: true,
  },
  over_18: {
    type: Boolean,
    required: true,
  },
  permalink: {
    type: String,
    required: true,
  },
  submission_link: {
    type: String,
    required: true,
  },
  saved: {
    type: Boolean,
    required: true,
  },
  score: {
    type: Number,
    required: true,
  },
  selftext: {
    type: String,
    required: true,
  },
  spoiler: {
    type: Boolean,
    required: true,
  },
  stickied: {
    type: Boolean,
    required: true,
  },
  subreddit: {
    type: String,
    required: true,
  },
  subreddit_id: {
    type: String,
    required: true,
  },
  title: {
    type: String,
    required: true,
  },
  upvote_ratio: {
    type: Number,
    required: true,
  },
  url: {
    type: String,
    required: true,
  },
});

module.exports = mongoose.model("Submission", submissionSchema);
